ext/webhook.go

```diff
@@ -27,13 +28,8 @@ type WebhookOpts struct {
 	SecretToken string
 }
 
-// GetListenAddr returns the local listening address, including port.
-func (w *WebhookOpts) GetListenAddr() string {
-	if w.Listen == "" {
-		w.Listen = "0.0.0.0"
+func (w *WebhookOpts) setDefaults() {
```

Comment:
What's more idiomatic; a `setDefaults` method, or a `GetListenNet` which handles the default case? 

I think I lean towards the second; I've been bitten by methods unexpectedly changing their input pointers before. What do you think?

Reply:
Not sure I understand about changing input pointers. `Webhookopts` stays at the same address, only the string changes.

About `GetListenNet`, could write a suggestion code ? I'm how you would use it.

Reply:
What I was trying to say is that by calling `setDefaults`, we end up changing the `opts` parameter in `StartServer`.

So, whoever calls `StartServer` has to be aware of the side effect that it will mutate the opts parameter it passed in, which seems like an unnecessary side effect.

Re: code, I just meant something like this, so the empty case is handled instead of defaulted.

```
func (w *WebhookOpts) GetListenNet() string {
	if w.ListenNet == "" {
		return "tcp"
	}
	return w.ListenNet
```

And then you would instead call `net.Listen(opts.GetListenNet()`, ...)` when starting the server.

Reply:
Argh, too used to all the other opts calls. Thought that one was passed as a pointer too, and didn't check to confirm.

Regardless, I still think it's nicer to avoid this pattern of "setting defaults", which could bite you if you forget to call it.

Reply:
In C, with an explicit `const` qualifier, it's easier to take a pointer while telling the user that we won't modify anything (even if it's not enforced per se, one can always cast const away).